.pipelines/cni/cilium/cilium-overlay-load-test-template.yaml

```diff
@@ -64,64 +72,62 @@ stages:
           - template: ../load-test-templates/pod-deployment-template.yaml
             parameters:
-              clusterName: ${{ parameters.clusterName }}
+              clusterName: ${{ parameters.clusterName }}-$(commitID)
               scaleup: ${CILIUM_SCALEUP}
               os: linux
               iterations: ${CILIUM_ITERATIONS}
               nodeCount: ${{ parameters.nodeCount }}
-  - stage: validate_state
-    dependsOn: pod_deployment
-    displayName: "Validate State"
-    jobs:
-      - job: validate_state
-        pool:
-          name: "$(BUILD_POOL_NAME_DEFAULT)"
-        steps:
           - template: ../load-test-templates/validate-state-template.yaml
             parameters:
-              clusterName: ${{ parameters.clusterName }}
-  - stage: restart_nodes
-    dependsOn: validate_state
-    displayName: "Restart Node"
-    jobs:
+              clusterName: ${{ parameters.clusterName }}-$(commitID)
       - job: restart_nodes
-        pool:
-          name: "$(BUILD_POOL_NAME_DEFAULT)"
+        displayName: "Restart Test"
+        dependsOn: deploy_pods
         steps:
           - template: ../load-test-templates/restart-node-template.yaml
             parameters:
-              clusterName: ${{ parameters.clusterName }}
+              clusterName: ${{ parameters.clusterName }}-$(commitID)
               nodeCount: ${{ parameters.nodeCount }}
               scaleup: ${CILIUM_SCALEUP}
-  - stage: validate_restart_state
-    dependsOn: restart_nodes
-    displayName: "Validate Restart State"
-    jobs:
-      - job: validate_restart_state
-        pool:
-          name: "$(BUILD_POOL_NAME_DEFAULT)"
-        steps:
           - template: ../load-test-templates/validate-state-template.yaml
             parameters:
-              clusterName: ${{ parameters.clusterName }}
+              clusterName: ${{ parameters.clusterName }}-$(commitID)
               restartCase: "true"
-  - stage: connectivity_tests
-    dependsOn: validate_restart_state
-    displayName: "Connectivity Tests"
-    jobs:
+              continueOnError: true
+      - job: restart_cns
+        displayName: "Restart and Validate CNS"
+        dependsOn: restart_nodes
+        steps:
+          - task: AzureCLI@1
+            inputs:
+              azureSubscription: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+              scriptLocation: "inlineScript"
+              scriptType: "bash"
+              addSpnToEnvironment: true
+              inlineScript: |
+                make -C ./hack/aks set-kubeconf AZCLI=az CLUSTER=${{ parameters.clusterName }}-$(commitID)
+                kubectl get pod -owide -A
+                echo "Validate pod IP assignment before CNS restart"
+                make test-validate-state
+                echo "restart CNS"
+                kubectl rollout restart ds azure-cns -n kube-system
+                kubectl rollout status ds azure-cns -n kube-system
+                sleep 15 # curl -ing after restart without a delay is flaky

```

Comment:
how did we land up on 15 ?
If CNS is taking time to restart, should we flag it to make sure we have not impacted the perf of cns.

Reply:
15s was arbitrary. There is no hard coded timeframe within CNS for it to be it up. It can get removed.